code/modules/awaymissions/redgate.dm

```diff
@@ -54,9 +55,24 @@
 	var/turf/ourturf = find_our_turf(M)		//Find the turf on the opposite side of the target
 	if(!ourturf.check_density(TRUE,TRUE))	//Make sure there isn't a wall there
 		M.unbuckle_all_mobs(TRUE)
-		M.stop_pulling()
-		playsound(src,'sound/effects/ominous-hum-2.ogg', 100,1)
-		M.forceMove(ourturf)		//Let's just do forcemove, I don't really want people teleporting to weird places if they have bluespace stuff
+		if(M.pulling)
+			var/atom/movable/pulled = M.pulling
+			M.stop_pulling()
+			playsound(src,'sound/effects/ominous-hum-2.ogg', 100,1)
+			M.forceMove(ourturf)
+			if(pulled.type in exceptions)
+				if(istype(pulled,/obj/structure/bed/roller))
+					var/obj/structure/bed/roller/R = pulled
+					for(var/mob/on_roller in R.buckled_mobs)
+						if(!on_roller.key || (on_roller.type in restrictions))
```

Comment:
Same for mob restrictions as I said above, but also, I'd suggest not making this check exclusive to roller beds, and instead all objects, since its possible more will be added later.

Reply:
my mistake, missed that buckled mobs applies to all movable, gonna change that up